tests/delegate_to_enum.rs

```diff
@@ -0,0 +1,62 @@
+use delegate::delegate;
+
+enum Enum {
+    A(A),
+    B(B),
+    C { v: C },
+}
+
+struct A {
+    val: usize,
+}
+
+impl A {
+    fn dbg_inner(&self) {
+        dbg!(self.val);
+    }
+}
+struct B {
+    val_a: String,
+}
+
+impl B {
+    fn dbg_inner(&self) {
+        dbg!(self.val_a.clone());
+    }
+}
+
+struct C {
+    val_c: f64,
+}
+
+impl C {
+    fn dbg_inner(&self) {
+        dbg!(self.val_c);
+    }
+}
+
+impl Enum {
+    delegate! {
+        to match self {
+            Enum::A(a) => a,
+            Enum::B(b) => b,
+            Enum::C { v: c } => c,
+        } {
+            fn dbg_inner(&self) ;
+        }
+    }
+}
+
+#[test]
+fn test_delegate_enum() {
+    let a = Enum::A(A { val: 1 });
+    a.dbg_inner();
```

Comment:
Please change `dbg_inner` so that it returns something (e.g. `u32`) and add some `asserts` here to check that the correct method is being called (`A::dbg_inner` will return `1`, `B::dbg_inner` will return `2` etc.).